src/Liquidator.sol

```diff
@@ -175,143 +167,156 @@ contract Liquidator is Owned, ILiquidator {
+
     /**
      * @notice Places a bid.
      * @param account The contract address of the Account being liquidated.
      * @param askedAssetAmounts Array with the assets-amounts the bidder wants to buy.
-     * @param endAuction Bool indicating if the bidder wants to end the auction.
+     * @param endAuction_ Bool indicating that the auction can be ended after the bid..
      * @dev We use a dutch auction: price of the assets constantly decreases.
+     * @dev The bidder is not obliged to set endAuction to True if the account is healthy after the bid,
+     * but they are incentivised to do so by earning an additional "auctionTerminationReward".
      */
-    function bid(address account, uint256[] memory askedAssetAmounts, bool endAuction) external {
-        // Check if the account is already in an auction.
+    function bid(address account, uint256[] memory askedAssetAmounts, bool endAuction_) external {
         AuctionInformation storage auctionInformation_ = auctionInformation[account];
         if (!auctionInformation_.inAuction) revert Liquidator_NotForSale();
 
-        // Calculate the current auction price.
-        uint256 askedShare = _calculateAskedShare(auctionInformation_, askedAssetAmounts);
-        uint256 price = _calculateBidPrice(auctionInformation_, askedShare);
+        // Calculate the current auction price of the assets being bought.
+        uint256 totalShare = _calculateTotalShare(auctionInformation_, askedAssetAmounts);
+        uint256 price = _calculateBidPrice(auctionInformation_, totalShare);
 
-        // Repay the debt of the account.
-        bool earlyTerminate = ILendingPool(auctionInformation_.creditor).auctionRepay(
-            auctionInformation_.startDebt, price, account, msg.sender
-        );
+        // Transfer an amount of "price" in "baseCurrency" to the LendingPool to repay the Accounts debt.
+        // The LendingPool will call a "transferFrom" from the bidder to the pool -> the bidder must approve the LendingPool.
+        // If the amount transferred would exceed the debt, the surplus is paid out to the Account Owner and earlyTerminate is True.
+        uint128 startDebt = auctionInformation_.startDebt;
+        bool earlyTerminate =
+            ILendingPool(auctionInformation_.creditor).auctionRepay(startDebt, price, account, msg.sender);
 
         // Transfer the assets to the bidder.
         IAccount(account).auctionBid(
             auctionInformation_.assetAddresses, auctionInformation_.assetIds, askedAssetAmounts, msg.sender
         );
 
-        // If all the debt is paid back, end the auction early.
-        // No need to do a health check for the account since it has no debt anymore.
+        // If all the debt is repaid, the auction must be ended, even if the bidder did not set endAuction to true.
         if (earlyTerminate) {
-            // Stop the auction
-            auctionInformation[account].inAuction = false;
+            // Happy flow: All debt is repaid.
```

Comment:
if earlyTerminate and not endAuction_, we never go into the settleliquidation flows -> no syncing of fees to liquidity providers